src/DoctrineAuditBundle/Routing/CustomLoader.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace DH\DoctrineAuditBundle\Routing;
+
+use Symfony\Bundle\FrameworkBundle\Routing\AnnotatedRouteControllerLoader;
+use Symfony\Component\Config\Loader\Loader;
+use Symfony\Component\Config\Loader\LoaderInterface;
+use Symfony\Component\Routing\RouteCollection;
+
+class CustomLoader extends Loader implements LoaderInterface
+{
+    private $annotationLoader;
+
+    private $configuration;
+
+    public function load($resource, ?string $type = null): RouteCollection
+    {
+        $routeCollection = new RouteCollection();
+        if (true === $this->configuration['enable_audit_controller']) {
+            $routeCollection = $this->annotationLoader->load('DH\DoctrineAuditBundle\Controller\AuditController');
+        }
+
+        return $routeCollection;
+    }
+
+    public function supports($resource, ?string $type = null): bool
+    {
+        return 'annotation' === $type;
```

Comment:
You should use an other name type for this loader to make it more specific to the bundle. Like "audit_route" or "extra_audit"